WiseUnpacker/SabreTools.Serialization.Deserializers/WiseSectionHeader.cs

```diff
@@ -87,39 +87,193 @@ public class WiseSectionHeader : BaseBinaryDeserializer<SectionHeader>
+                        {
+                            
+                            byte checkForZero = 0x00;
+                            while (checkForZero == 0x00)
+                            {
+                                checkForZero = data.ReadByteValue();
+                            }
+                            data.Seek(data.Position - 1, 0);
+                            endNow = true;
+                            break;
+                        }
+                        currentByte = header.StringValues[counter];
+                        
+                        // 0x01 followed by one more 0x01 seems to indicate to skip 2 null bytes, but 0x01 followed by
+                        // three more 0x01 seems to indicate an unspecified length of null bytes that must be skipped.
+                        // It has already been observed it mean 22 or 27 between 2 samples.
+                        
+                        // If you encounter a null byte in the actual pre-string byte array, it seems to always be
+                        // after you've read all the strings successfully.
+                        if (currentByte == 0x00) // this should never happen
+                        {
+                            endNow = true; 
+                            break;
+                        }
+                        else if (currentByte != 0x01)
+                        {
+                            byte checkForZero = 0x00;
+                            while (checkForZero == 0x00)
+                            {
+                                checkForZero = data.ReadByteValue();
+                            }
 
-            header.UnknownValue18 = data.ReadUInt32LittleEndian();
-            if (headerLength == 18)
-                return header;
+                            data.Seek(data.Position - 1, 0);
+                            break;
+                        }
+                        else
+                        {
+                            oneCount++;
+                        }
+                        counter++;
+                    }
+                    if (oneCount == 4) 
+                    {
+                        languageSection = true;
+                    }
+                }
+                else if (currentByte == 0x00) // this should never happen
+                {
+                    endNow = true;
+                }
+                if (endNow == true)
+                    break;
+                byte[] currentString = data.ReadBytes(currentByte); // System.Text.Encoding.ASCII.GetString(currentString);
+                stringList.AddRange(currentString);
+                counter++;
+                if (languageSection)
+                {
+                    languageSectionCounter++;
+                }
+            }
+            
+            // Strings stored as byte array since one "string" can contain multiple null-terminated strings.
+            header.Strings = stringList.ToArray(); 
```

Comment:
The comment about the model would impact how this would be populated. For example, each string would be read into a separate `byte[]` and then those would be added to a `List<byte[]>` which could then be assigned to `header.Strings` by doing `[.. stringList]`.